easy-ui-react/src/TextField/TextField.module.scss

```diff
@@ -0,0 +1,274 @@
+@use "../styles/common" as *;
+
+.root {
+  @include component-token(
+    "textfield",
+    "border_radius",
+    design-token("shape.border_radius.md")
+  );
+  @include component-token(
+    "textfield",
+    "box_shadow",
+    design-token("shadow.input")
+  );
+  @include component-token(
+    "textfield",
+    "color.text.resting",
+    theme-token("color.text")
+  );
+  @include component-token(
+    "textfield",
+    "color.text.gray.resting",
+    theme-token("color.text.gray.resting")
+  );
+  @include component-token(
+    "textfield",
+    "color.text.disabled",
+    theme-token("color.text.disabled")
+  );
+  @include component-token(
+    "textfield",
+    "color.text.engaged",
+    theme-token("color.text.engaged")
+  );
+  @include component-token(
+    "textfield",
+    "color.text.danger",
+    theme-token("color.text.danger")
+  );
+  @include component-token(
+    "textfield",
+    "color.border.resting",
+    theme-token("color.border.resting")
+  );
+  @include component-token(
+    "textfield",
+    "color.border.engaged",
+    theme-token("color.border.engaged")
+  );
+  @include component-token(
+    "textfield",
+    "color.border.danger",
+    theme-token("color.border.danger")
+  );
+  @include component-token(
+    "textfield",
+    "color.background",
+    theme-token("color.background")
+  );
+  @include component-token(
+    "textfield",
+    "color.background.disabled",
+    theme-token("color.background.disabled.subtle")
+  );
+  @include component-token("textfield", "max-height-sm", 32px);
+  @include component-token(
+    "textfield",
+    "z-index",
+    design-token("z_index.input_icon")
+  );
+}
+
+.inputLabelContainer {
+  display: flex;
+  flex-direction: column;
+}
+
+.inputIconContainer {
+  display: flex;
+  align-items: center;
+  position: relative;
+}
+
+.input {
+  @include font-style("body1");
+  color: component-token("textfield", "color.text.resting");
+  background-color: component-token("textfield", "color.background");
+  border: design-token("shape.border_width.1") solid
+    component-token("textfield", "color.border.resting");
+  border-radius: component-token("textfield", "border_radius");
+  padding: calc(
+    #{design-token("space.1.5")} - #{design-token("shape.border_width.1")}
+  );
+  outline: none;
+  position: relative;
+  width: 100%;
+
+  &:focus,
+  &:active {
+    color: component-token("textfield", "color.text.engaged");
+    border: design-token("shape.border_width.1") solid
+      component-token("textfield", "color.border.engaged");
```

Comment:
**suggestion:** if the border size and style doesn't change from what's set on the parent selector, it'd be preferable to only update the part that changes. in this case i believe it's just `border-color`. that way you only have one place where the primary border configuration is set

Reply:
ah yeah, good call